app/src/main/java/com/myperishableplanner/v21001/dao/IItemDAO.kt

```diff
@@ -3,8 +3,9 @@ package com.myperishableplanner.v21001.dao
 import com.myperishableplanner.v21001.dto.Item
 import retrofit2.http.GET
 import retrofit2.Call
+import retrofit2.http.Query
 
 interface IItemDAO {
-    @GET("fdc/v1/foods/list?api_key=but9Hwci1MFrkr6A3RXH0o7kgO9gjbHfgl01U2JC&query=apple")
-    fun getAllItems() : Call<List<Item>>
+    @GET("fdc/v1/foods/list")
+    fun getAllItems(@Query("query") searchTxt: String, @Query("api_key") type: String? = "but9Hwci1MFrkr6A3RXH0o7kgO9gjbHfgl01U2JC"): Call<List<Item>>
```

Comment:
ItemDAO now has a searchTxt string parameter which set the 'query' value in the API calls query string instead of the hard coded apple value. I also deleted the Milk and Waterman DAO's as they are simple copies of this DAO with a different hardcoded food value. You could also move this api key to local.properties and reference it instead of making it public here.

Reply:
A few cautions here:
- This looks like a change in functionality.
- getAllItems isn't getting all items if there is a filter applied.
- Does the API support this query?  If not, it's misleading.
I recommend setting this up as a new function, and leave getAllItems as it were: getting all items.

Reply:
It wasn't getting all items. It was getting all "apples",  @GET("fdc/v1/foods/list?api_key=but9Hwci1MFrkr6A3RXH0o7kgO9gjbHfgl01U2JC&**query=apple**"). 
Then they had an exact copy this code except with Milk, Watermelon, etc. The api works this way. I tested it.